Revert commit " build: Do not compile uninstalled GIR into typelib "

This causes the typelibs to be installed again.

author: Alexander Grund (TU Dresden)
diff -ur gobject-introspection-1.80.1-orig/gir/meson.build gobject-introspection-1.80.1/gir/meson.build
--- gobject-introspection-1.80.1-orig/gir/meson.build	2025-02-13 16:58:34.700788731 +0100
+++ gobject-introspection-1.80.1/gir/meson.build	2025-02-13 17:06:59.695300274 +0100
@@ -12,9 +12,6 @@
   'xrandr-1.3.gir',
   'win32-1.0.gir',
 ]
-
-uninstalled_gir_files = []
-
 # Copy gir files to build directory to have them all in a single place.
 # This is needed when gobject-introspection is a subproject because Meson
 # will add --includedir pointing to build directory automatically.
@@ -255,7 +252,7 @@
   ]
 )
 
-uninstalled_gir_files += glib_gir
+gir_files += glib_gir
 
 # GObject
 gobject_files = []
@@ -323,7 +320,7 @@
   ]
 )
 
-uninstalled_gir_files += gobject_gir
+gir_files += gobject_gir
 
 # GModule
 gmodule_files = []
@@ -359,7 +356,7 @@
 # out of date.
 gmodule_files += files('gmodule-2.0.c')
 
-uninstalled_gir_files += custom_target('gir-gmodule',
+gir_files += custom_target('gir-gmodule',
   input: gmodule_files,
   output: 'GModule-2.0.gir',
   depends: [glib_gir, gir_giscanner_pymod, gmodule_gir_dep, gdump],
@@ -474,7 +471,7 @@
   ]
 )
 
-uninstalled_gir_files += gio_gir
+gir_files += gio_gir
 
 # GIRepository
 
@@ -515,19 +512,17 @@
 
 typelibs = []
 if get_option('gi_cross_binary_wrapper') != ''
-  gircompiler_command = [get_option('gi_cross_binary_wrapper'), gircompiler.full_path(), ]
+    gircompiler_command = [get_option('gi_cross_binary_wrapper'), gircompiler.full_path(), ]
 else
-  gircompiler_command = [gircompiler, ]
+    gircompiler_command = [gircompiler, ]
 endif
 
-gircompiler_command += [
-  '-o', '@OUTPUT@',
-  '@INPUT@',
-  '--includedir', meson.current_build_dir(),
-  '--includedir', meson.current_source_dir(),
+gircompiler_command += [ '-o', '@OUTPUT@', '@INPUT@',
+              '--includedir', meson.current_build_dir(),
+              '--includedir', meson.current_source_dir(),
 ]
 
-foreach gir: gir_files
+foreach gir : gir_files
   typelibs += custom_target('generate-typelib-@0@'.format(gir).underscorify(),
     input: gir,
     output: '@BASENAME@.typelib',
@@ -538,15 +533,6 @@
   )
 endforeach
 
-foreach gir: uninstalled_gir_files
-  typelibs += custom_target('generate-typelib-@0@'.format(gir).underscorify(),
-    input: gir,
-    output: '@BASENAME@.typelib',
-    depends: [gobject_gir, gircompiler, ],
-    command: gircompiler_command,
-  )
-endforeach
-
 rnc2rng = find_program('rnc2rng', native: true, required: false)
 if rnc2rng.found()
   validate_gir = find_program('../misc/validate-gir.py')
@@ -562,16 +548,4 @@
       workdir: meson.current_build_dir(),
     )
   endforeach
-
-  foreach gir: uninstalled_gir_files
-    gir_name = gir.full_path().split('/')[-1]
-    test('validate-@0@'.format(gir_name).underscorify(),
-      python,
-      args: [
-        validate_gir.full_path(),
-        '--schema=@0@'.format(meson.current_source_dir() / '../docs/gir-1.2.rnc'),
-        gir,
-      ],
-    )
-  endforeach
 endif
